spring-boot-autoconfigure/src/main/java/org/springframework/boot/autoconfigure/amqp/RabbitAnnotationDrivenConfiguration.java

```diff
@@ -65,12 +68,19 @@ public SimpleRabbitListenerContainerFactoryConfigurer rabbitListenerContainerFac
 
 	@Bean
 	@ConditionalOnMissingBean(name = "rabbitListenerContainerFactory")
-	public SimpleRabbitListenerContainerFactory rabbitListenerContainerFactory(
-			SimpleRabbitListenerContainerFactoryConfigurer configurer,
+	public AbstractRabbitListenerContainerFactory<?> rabbitListenerContainerFactory(
```

Comment:
I started down that route, but it didn't work as I expected.

I had `@ConditionalOnProperty( ... havingValue = "simple")` 

and `@ConditionalOnProperty( ... havingValue = "direct")`

...it didn't work without explicitly setting the property - but I now see there's a `matchIfMissing` attribute ... doh.

Will rework.